shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/controller/ExecutionApi.java

```diff
@@ -0,0 +1,103 @@
+/**
+
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.vip.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.shanoir.ng.shared.core.model.IdName;
+import org.shanoir.ng.shared.exception.EntityNotFoundException;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.shanoir.ng.shared.exception.SecurityException;
+import org.shanoir.ng.vip.dto.ExecutionCandidateDTO;
+import org.shanoir.ng.vip.dto.VipExecutionDTO;
+import org.shanoir.ng.vip.monitoring.model.ExecutionStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.io.IOException;
+
+/**
+ * @author Alae Es-saki
+ */
+@Tag(name = "VIP execution", description="Proxy API for VIP /executions")
+@RequestMapping("/vip/execution")
+public interface ExecutionApi {
+
+    @Operation(summary = "Creates an execution inside VIP and return the monitoring id and name", description = "Creates the ressources and path control in shanoir before creating an execution in VIP, then return the running execution monitoring", tags={  })
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Execution successfully created response."),
+            @ApiResponse(responseCode = "403", description = "forbidden"),
+            @ApiResponse(responseCode = "500", description = "unexpected error"),
+            @ApiResponse(responseCode = "503", description = "error from VIP API")})
+    @RequestMapping(value = "/",
+            produces = {"application/json"},
+            consumes = {"application/json"},
+            method = RequestMethod.POST)
+    ResponseEntity<IdName> createExecution(
```

Comment:
Hi, should we put at least a role protection on the API methods?

Reply:
The roles are checked in the method, but we should actually put it here into a specific class

Reply:
In fact, as documented [here](https://github.com/fli-iam/shanoir-ng/wiki/Roles-and-rights-specification), the required rights on datasets included in execution is `CAN_ADMINISTRATE`.

I've corrected this and extracted the check into a `checkRightsForExecution` method. If relevant, this will throw a proper unauthorized RestServiceException with an explicit message.